index.js

```diff
@@ -344,22 +350,15 @@ let rcl = {
    * this folder will be overwritten after the execution.
    * @return {Promise<undefined>} A Promise.
    */
-  regenerateAll() {
+  async regenerateAll() {
     // This will trigger to regenerate all the JS structs used for messages and services,
     // to overwrite the existing ones although they have been created.
-    debug('Begin regeneration of JavaScript code from ROS IDL files.');
-    return new Promise((resolve, reject) => {
-      generator
-        .generateAll(true)
-        .then(() => {
-          tsdGenerator.generateAll(); // create interfaces.d.ts
-          debug('Finish regeneration.');
-          resolve();
-        })
-        .catch((e) => {
-          reject(e);
-        });
-    });
+    debug('Begin regeneration of JavaScript code from ROS IDL files...');
+
+    await generator.generateAll(true);
```

Comment:
Does this need to happen in sequence or could this also be done "concurrently" with `Promise.all([...])`?

Reply:
The ts generator depends on the JS files generated by `generator.generateAll(true)`, @wayneparrott please help to take a look, thanks!

Reply:
@minggangw @felixdivo Yes, message generation is a 2 phase process: 1) generate js message files, 2) generate the interfaces.d.ts declaration file from the js message files.

Reply:
Okay, thanks. Then this PR should be done. I'll just add a comment about the above.

Reply:
```suggestion
    # generate the messages and type declaration, which must be done in sequence
    await generator.generateAll(true);
```